src/storage/txn/store.rs

```diff
@@ -215,6 +227,36 @@ impl<'a> SnapshotStore<'a> {
         }
         Ok(results)
     }
+
+    pub fn reverse_scan(&self, key: Key, limit: usize) -> Result<Vec<Result<KvPair>>> {
+        let mut results = vec![];
+        let mut key = key;
+        let txn = MvccSnapshot::new(self.snapshot.as_ref(), self.start_ts);
+        while results.len() < limit {
+            key = match try!(self.snapshot.reverse_seek(&key)) {
+                Some((k, _)) => try!(extract_key(&k)),
+                None => break,
+            };
+            match txn.get(&key) {
+                Ok(Some(value)) => results.push(Ok((key.raw().to_owned(), value))),
+                Ok(None) => {}
+                e @ Err(MvccError::KeyIsLocked { .. }) => {
+                    results.push(Err(Error::from(e.unwrap_err())))
+                }
+                Err(e) => return Err(e.into()),
+            };
+        }
+        Ok(results)
+    }
+}
+
+/// `extract_key` discards the 8 bytes mvcc version, returns the real key.
+fn extract_key(key: &[u8]) -> Result<Key> {
```

Comment:
@BusyJay I think it is improper to do that. We should only call this functions when we know the key is mvcc-encoded.